### PR TITLE
ATO-1064: add audit log for bsid sign out

### PR DESCRIPTION
**Pull request by GHSwallow**
## What
Adds audit log for a user needing to authenticate as a result of having an invalid "bsid". This is only logged if the "bsid" is the _only_ reason for a new authentication journey, so not logged if both "gs" and "bsid" are missing, for example. This means only the journeys affected by this change will be logged.

Original ticket to make a decision on this auditting: https://govukverify.atlassian.net/browse/ATO-969
Event catalogue update: https://github.com/govuk-one-login/event-catalogue/pull/234
Slack discussions: https://gds.slack.com/archives/C021FD820N6/p1724763497125069 and https://gds.slack.com/archives/C05811D4JMR/p1724766923917489

## How to review

1. Code Review

### CODE REVIEW

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java [hunk 1]
```diff
@@ -409,28 +429,15 @@ public APIGatewayProxyResponseEvent authoriseRequestHandler(
                 && browserSessionIdFromSession.isPresent()
                 && !Objects.equals(browserSessionIdFromSession, browserSessionIdFromCookie)) {
             sessionWithValidBrowserSessionId = Optional.empty();
+            auditService.submitAuditEvent(
+                    OidcAuditableEvent.AUTHORISATION_INITIATED,
+                    authRequest.getClientID().getValue(),
+                    user,
+                    pair("new_authentication_required", true));
         } else {
```
Comment (billrdunn) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java:437]: Is this logically the same, without the `else`?

```
        Optional<Session> sessionWithValidBrowserSessionId = session;
        if (configurationService.isBrowserSessionCookieEnabled()
                && configurationService.isSignOutOnBrowserCloseEnabled()
                && browserSessionIdFromSession.isPresent()
                && !Objects.equals(browserSessionIdFromSession, browserSessionIdFromCookie)) {
            sessionWithValidBrowserSessionId = Optional.empty();
            auditService.submitAuditEvent(
                    OidcAuditableEvent.AUTHORISATION_INITIATED,
                    authRequest.getClientID().getValue(),
                    user,
                    pair("new_authentication_required", true));
        }
```

Comment (GHSwallow) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java:437]: Yea, good spot!

**Comment by billrdunn** [comment]
I can't remember exactly how it works - is there a way of confirming the audit events are being sent and received successfully in a lower env before releasing to prod?

**Comment by isaac-GDS** [comment]
> I can't remember exactly how it works - is there a way of confirming the audit events are being sent and received successfully in a lower env before releasing to prod?

Think Staging is where we test audit logs TXMA. We can pause in staging to see if the logs are going through.

**Comment by GHSwallow** [comment]
`isSignOutOnBrowserCloseEnabled` is currently only turned on up to staging, so we can test without pausing I believe

**Review by isaac-GDS — APPROVED** [review]

**Pull request merged**